src/lib.rs

```diff
@@ -487,6 +492,37 @@ pub enum LogLevel {
     Warning,
 }
 
+/// Extend the [`EventSender`] functionality to work conviently with a [`Wallet`](bdk_wallet).
+#[cfg(feature = "wallet")]
+pub trait EventSenderExt {
+    /// Add all revealed scripts to the node to monitor.
+    fn add_revealed_scripts<'a>(
+        &'a self,
+        wallet: &'a bdk_wallet::Wallet,
+    ) -> FutureResult<(), kyoto::ClientError>;
+}
+
+impl EventSenderExt for EventSender {
+    fn add_revealed_scripts<'a>(
+        &'a self,
+        wallet: &'a bdk_wallet::Wallet,
+    ) -> FutureResult<(), kyoto::ClientError> {
+        async fn _add_revealed(
+            sender: &EventSender,
+            wallet: &bdk_wallet::Wallet,
+        ) -> Result<(), kyoto::ClientError> {
+            for keychain in [KeychainKind::External, KeychainKind::Internal] {
+                let scripts = wallet.spk_index().revealed_keychain_spks(keychain);
+                for (_, script) in scripts {
+                    sender.add_script(script).await?;
+                }
+            }
+            Ok(())
+        }
+        Box::pin(_add_revealed(&self, wallet))
```

Comment:
Also caught by CI but apparently we allow warnings. We could change clippy in CI to `cargo clippy --all-targets -- -Dwarnings`

```suggestion
        Box::pin(_add_revealed(self, wallet))
```

Reply:
Yeah we can follow up with that patch to CI